### ISSUE TITLE
Error for Hadopp version 3.2.1

### ISSUE THREAD

**Issue by etashsingh** [marcelmay/hadoop-hdfs-fsimage-exporter#34]
@marcelmay 
We are trying to run the exporter for our Hadoop v3.2.1 fsimage. 
We are getting this error:
```
2020-09-07 01:34:04,110 [pool-1-thread-1] WARN  org.apache.hadoop.util.NativeCodeLoader  - Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
2020-09-07 01:34:04,113 [pool-1-thread-1] ERROR de.m3y.prometheus.exporter.fsimage.FsImageUpdateHandler  - Can not load FSImage
java.lang.NullPointerException
	at org.apache.hadoop.io.compress.CompressionCodecFactory.getCodecClasses(CompressionCodecFactory.java:124)
	at org.apache.hadoop.io.compress.CompressionCodecFactory.<init>(CompressionCodecFactory.java:180)
	at org.apache.hadoop.hdfs.server.namenode.FSImageCompression.createCompression(FSImageCompression.java:100)
	at org.apache.hadoop.hdfs.server.namenode.FSImageUtil.wrapInputStreamForCompression(FSImageUtil.java:88)
	at de.m3y.hadoop.hdfs.hfsa.core.FSImageLoader.loadSection(FSImageLoader.java:156)
	at de.m3y.hadoop.hdfs.hfsa.core.FSImageLoader.load(FSImageLoader.java:186)
	at de.m3y.prometheus.exporter.fsimage.FsImageUpdateHandler.loadFsImage(FsImageUpdateHandler.java:212)
	at de.m3y.prometheus.exporter.fsimage.FsImageUpdateHandler.onFsImageChange(FsImageUpdateHandler.java:191)
	at de.m3y.prometheus.exporter.fsimage.FsImageWatcher.run(FsImageWatcher.java:63)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

We have tried for a lower version where this is working fine, but we need to use version 3.2.1 in any case.
How can we overcome this issue?

**Comment by marcelmay** [comment]
@etashsingh , can you tell me your ```dfs.image.compression.codec``` configuration setting? I'll try to reproduce this.

**Comment by etashsingh** [comment]
@marcelmay, here is the configuration for dfs.image.compression.codec
```
<property>
        <name>dfs.image.compress</name>
        <value>true</value>
    </property>
    <property>
        <name>dfs.image.compression.codec</name>
        <value>org.apache.hadoop.io.compress.DefaultCodec</value>
    </property>
```

**Comment by marcelmay** [comment]
@etashsingh , I was able to reproduce the error with a test case including a compressed fsimage, and added a fix in the dependent hfsa lib (```FSImageLoader.loadSection``` in the stacktrace).

For fixing the exporter I will have to update the hfsa lib and do some more tests, and will probably do a release this Thursday.

**Comment by marcelmay** [comment]
Thx alot, @etashsingh and let me know if newly released version 1.4 works for you.

**Comment by etashsingh** [comment]
@marcelmay thanks for providing a quick fix. We have tested and it is working as expected now with version 3.2.1
